nexus/src/app/address_lot.rs

```diff
@@ -19,23 +22,43 @@ use std::sync::Arc;
 use uuid::Uuid;
 
 impl super::Nexus {
+    pub fn address_lot_lookup<'a>(
+        &'a self,
+        opctx: &'a OpContext,
+        address_lot: NameOrId,
+    ) -> LookupResult<lookup::AddressLot<'a>> {
+        match address_lot {
+            NameOrId::Id(id) => {
+                let lot = LookupPath::new(opctx, &self.db_datastore)
+                    .address_lot_id(id);
+                Ok(lot)
+            }
+            NameOrId::Name(name) => {
+                let lot = LookupPath::new(opctx, &self.db_datastore)
+                    .address_lot_name_owned(name.into());
+                Ok(lot)
+            }
+        }
+    }
```

Comment:
I need to document this, but a typical pattern you'll see for every resource in the app layer is a lookup function that looks similar to this. It takes either the name or id (for top level resources) or a selector (for resources nested deeper in the hierarchy) and returns a lookup. The lookup that's returned from here is all that's intended to be passed to other layers of the API.